spec/src/main/asciidoc/Structure and format.adoc

```diff
@@ -0,0 +1,81 @@
+== Structure and format
+
+=== Terms
+
+.Terms and definitions
+Configuration:: An optional _top-level value_, with a collection of zero or more nested _configuration entries_ (organized by _configuration key_).
+  Also informally known as a "configuration tree node" or "configuration node".
+  A configuration with no value and no nested entries is considered to be an _empty configuration_.
+
+Configuration entry:: An _configuration key_ and _configuration value_ combination.
+
+Configuration key:: A non-empty string which uniquely identifies a configuration entry within a single nesting level of a configuration.
+  A configuration key *must not* contain embedded newlines or other control characters.
+  A configuration key string *must* be normalized using https://www.unicode.org/reports/tr15/#Norm_Forms)[Unicode Normalization Form C (NFC)].
+
+Configuration path:: A sequence of _configuration keys_ which uniquely identify a descendant _configuration entry_ within a configuration, at an arbitrary level of nesting.
```

Comment:
How about `Configuration Tree` or `Configuration Cluster` instead of `Configuration path`, as at first I thought you meant the path to the config sources?

Reply:
Some suggestions were offered up in the call. Vote each one, and add your own!

Reply:
`Configuration key` as a sequence of `names`

Reply:
`Configuration identifier` as a sequence of `keys`